src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java

```diff
@@ -0,0 +1,580 @@
+    public static PublicShareDialogFragment newInstanceToCreate(OCFile fileToShare, Account account) {
+        PublicShareDialogFragment publicShareDialogFragment = new PublicShareDialogFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(ARG_FILE, fileToShare);
+        args.putParcelable(ARG_ACCOUNT, account);
+
+        publicShareDialogFragment.setArguments(args);
+        return publicShareDialogFragment;
+    }
+
+    /**
+     * Update an instance of PublicShareDialogFragment, providing fileToShare, publicShare and
+     * account as arguments.
+     *
+     * Dialog shown this way is intended to UPDATE an existing public share.
+     *
+     * @param   publicShare           Public share to update.
+     */
+    public static PublicShareDialogFragment newInstanceToUpdate(OCFile fileToShare,
+                                                                OCShare publicShare,
+                                                                Account account
+                                                                ) {
+        PublicShareDialogFragment publicShareDialogFragment = new PublicShareDialogFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(ARG_FILE, fileToShare);
+        args.putParcelable(ARG_SHARE, publicShare);
+        args.putParcelable(ARG_ACCOUNT, account);
+        publicShareDialogFragment.setArguments(args);
+        return publicShareDialogFragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments() != null) {
+            mFile = getArguments().getParcelable(ARG_FILE);
+            mAccount = getArguments().getParcelable(ARG_ACCOUNT);
+            mPublicShare = getArguments().getParcelable(ARG_SHARE);
+        }
+
+        if (mFile == null && mPublicShare == null) {
+            throw new IllegalStateException("Both ARG_FILE and ARG_SHARE cannot be NULL");
+        }
+
+        setStyle(DialogFragment.STYLE_NO_TITLE, 0);
+
+        refreshCapabilitiesFromDB();
+    }
+
+    private boolean updating() {
+        return (mPublicShare != null);
+    }
+
+    private boolean isSharedFolder() {
+        return (
+            (mFile != null && mFile.isFolder()) ||
+            (mPublicShare != null && mPublicShare.isFolder())
+        );
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.share_public_dialog, container, false);
+
+        // Show or hide edit permission section
+        if (isSharedFolder()){
+            getEditPermissionSection(view).setVisibility(View.VISIBLE);
+        } else {
+            getEditPermissionSection(view).setVisibility(View.GONE);
+        }
+
+        // Show default date enforced by the server, if any
+        if (mCapabilities.getFilesSharingPublicExpireDateDays() > 0) {
```

Comment:
Besides, all this block is only meaningful for "create mode", right? I'd rather use an ``` if (!updating) ... else ```  block instead of let this go all the times and later run the ```updating()```  if needed.